charon/src/logger.rs

```diff
@@ -11,130 +10,86 @@ pub fn initialize_logger() {
             return;
         }
     }
-    use env_logger::fmt::style;
-    use env_logger::{Builder, Env};
-    use std::io::Write;
 
-    // Create a default environment, by using the environment variables.
-    // We do this to let the user choose the log level (i.e.: trace,
-    // debug, warning, etc.)
-    let env = Env::default();
-    // If the log level is not set, set it to "info"
-    let env = env.default_filter_or("info");
-
-    // Initialize the log builder from the environment we just created
-    let mut builder = Builder::from_env(env);
-
-    // Modify the output format - we add the line number
-    builder.format(|buf, record| {
-        // Retreive the path (CRATE::MODULE) and the line number
-        let path = record.module_path().unwrap_or_default();
-        let line = record.line().unwrap_or_default();
-
-        // Style for the brackets (change the color)
-        let brackets_style = style::RgbColor(120, 120, 120).on_default();
-        let level_style = buf.default_level_style(record.level()); // Print the level with colors
-
-        write!(buf, "{brackets_style}[{brackets_style:#}")?;
-        write!(
-            buf,
-            "{level_style}{}{level_style:#} {path}:{line}",
-            record.level(),
-        )?;
-        write!(buf, "{brackets_style}]{brackets_style:#} ")?;
-        writeln!(buf, "{}", record.args())?;
-        Ok(())
-    });
-
-    builder.init();
-
-    // Also set up `tracing` so we get tracing data from `hax`.
     use tracing_subscriber::prelude::*;
-    let subscriber = tracing_subscriber::Registry::default()
+    tracing_subscriber::registry()
         .with(tracing_subscriber::EnvFilter::from_default_env())
         .with(
-            tracing_tree::HierarchicalLayer::new(2)
+            tracing_tree::HierarchicalLayer::new(1)
                 .with_ansi(true)
-                .with_indent_amount(1)
-                .with_indent_lines(true),
-        );
-    tracing::subscriber::set_global_default(subscriber).unwrap();
+                .with_indent_lines(true)
+                .with_bracketed_fields(true)
+                .with_timer(tracing_tree::time::Uptime::default()),
```

Comment:
Do we print the local time of the machine, or the time elapsed since the start of the execution?

Reply:
Each function call prints the time elapsed during that function call